File: Python/8py 1/define_suit.py
"""
|--------------------------------------------------------------------------
| Define a card suit
|--------------------------------------------------------------------------
|
"""

def define_suit(card):
   if 'C' in card: return 'clubs'
   if 'S' in card: return 'spades'
   if 'D' in card: return 'diamonds'
   if 'H' in card: return 'hearts'


def define_suit(card):
   return {'C':'clubs', 'S':'spades', 'D':'diamonds', 'H':'hearts'}[card[-1]]


def define_suit(card):
   return {'C':'clubs', 'D':'diamonds', 'H':'hearts', 'S':'spades'}.get(card[-1])


print(define_suit('3C'), 'clubs')
print(define_suit('QS'), 'spades')
print(define_suit('9D'), 'diamonds')
print(define_suit('JH'), 'hearts')

File: JavaScript/8js 1/defineSuit.js
/*
|--------------------------------------------------------------------------
| Define a card suit
|--------------------------------------------------------------------------
|
*/

function defineSuit(card) {
   return card[card.length - 1] == "♠"
      ? "spades"
      : card[card.length - 1] == "♦"
      ? "diamonds"
      : card[card.length - 1] == "♥"
      ? "hearts"
      : "clubs";
}


function defineSuit(card) {
   return {
      "♣": "clubs",
      "♦": "diamonds",
      "♥": "hearts",
      "♠": "spades",
   }[card.substr(-1)];
}


function defineSuit(card) {
   if (card.includes("♥")) return "hearts";
   if (card.includes("♦")) return "diamonds";
   if (card.includes("♣")) return "clubs";
   if (card.includes("♠")) return "spades";
}


console.log(defineSuit("Q♠"), "spades");
console.log(defineSuit("9♦"), "diamonds");
console.log(defineSuit("J♥"), "hearts");
console.log(defineSuit("3♣"), "clubs");